Mods/interface/PuppiMod.h

```diff
@@ -57,17 +53,26 @@ namespace mithep
       void SetDump( Bool_t dump )                            { fDumpingPuppi = dump;           }
 
     protected:
-      void                  SlaveBegin();
-      void                  SlaveTerminate();
-      void                  Process();
+      void                  SlaveBegin() override;
+      void                  SlaveTerminate() override;
+      void                  Process() override;
+
+      enum ParticleType {
+        kChargedPrimary = 1,
+        kChargedPU,
+        kNeutralCentral,
+        kNeutralForward
+      };
+
+      ParticleType GetParticleType( const PFCandidate *cand, Vertex const* ) const;
```

Comment:
These functions are moved to private as discussed above. Additionally, always declare a member function as const if it does not modify the class member variable. This improves code readability because now the reader knows (without looking at the implementation) that this function is harmless.